src/cpp/session/modules/SessionFind.cpp

```diff
@@ -975,20 +977,44 @@ struct ReplaceOptions
       replacePattern("")
    {}
 
-   ReplaceOptions(std::string replace, bool useGitIgnore) :
+   ReplaceOptions(std::string replace) :
       empty(false),
       preview(false),
-      useGitIgnore(useGitIgnore),
       replacePattern(replace)
    {}
 
    bool empty;
    bool preview;
-   bool useGitIgnore;
 
    const std::string replacePattern;
 };
 
+core::Error createGitExclusionString(const FilePath& dirPath, std::string* pResultString)
+{
+   shell_utils::ShellCommand cmd("git");
+   cmd << "-C";
+   cmd << string_utils::utf8ToSystem(dirPath.getAbsolutePath());
+   cmd << "ls-files";
+   cmd << "-i";
+   cmd << "--exclude-standard";
+
+   core::system::ProcessResult result;
+   core::system::ProcessOptions options;
+
+   core::Error error = runCommand(cmd,
+                                  options,
+                                  &result);
+   if (error)
+      return error;
+
+   *pResultString = result.stdOut;
+   size_t splitAt;
+   while ((splitAt = pResultString->find('\n')) != pResultString->npos)
```

Comment:
Would it be simpler to use `boost::algorithm::replace_all` here?

Reply:
Yes, thank you!